src/plugins/intel_cpu/src/nodes/scaled_attn.cpp

```diff
@@ -1916,6 +1928,8 @@ ov::element::Type ScaledDotProductAttention::getRuntimePrecision() const {
     // bf16 should be enabled only when platform supports
     if (rtPrecision == ov::element::bf16 && ov::with_cpu_x86_bfloat16()) {
         rtPrecision = ov::element::bf16;
+    } else if (rtPrecision == ov::element::f16 && hasHardwareSupport(ov::element::f16)) {
+        rtPrecision = ov::element::f16;
     } else {
```

Comment:
question here: on a platform which supports both bf16 and f16, will bf16 always be first priority? Which is better in terms of performance?

Reply:
There is no implicit priority when first checking bf16, here is trying to set the kvcache type based on the input type to avoid potential reorder.